src/Adapter/Translations/TranslationRouteFinder.php

```
@@ -203,7 +204,10 @@ public function findRouteParameters(ParameterBag $query)
     private function isModuleUsingNewTranslationSystem($moduleName)
     {
         $module = $this->moduleRepository->getInstanceByName($moduleName);
+        if (Validate::isLoadedObject($module)) {
+            return $module->isUsingNewTranslationSystem();
+        }
 
-        return $module->isUsingNewTranslationSystem();
+        return false;
```

This should throw an Exception if the module is not valid, not silently return false.

There should also be a try/catch block in TranslationsController::modifyTranslationsAction with a flash error message and a redirection to the overview page.

---

Hello @eternoendless Thanks !

The legacy controller is still present for modules not used by the new translation system. We agree that `AdminTranslationsController` (legacy) will also be impacted?

I do not think that modifs only at this level `TranslationsController :: modifyTranslationsAction` can cover all impacted zones.

What do you think ?